zerver/tests/test_management_commands.py

```diff
@@ -57,3 +59,40 @@ def test_check_if_command_post_request_to_url_with_fixture(self,
         self.assertTrue(ujson_mock.loads.called)
         self.assertTrue(open_mock.called)
         client.post.assert_called_once_with(self.url, {}, content_type="application/json")
+
+class TestGenerateRealmCreationLink(TestCase):
+    COMMAND_NAME = "generate_realm_creation_link"
+
+    def test_generate_link_and_create_realm(self):
+        username = "user1"
+        domain = "test.com"
+        email = "user1@test.com"
+        generated_link = generate_realm_creation_url()
+
+        with self.settings(OPEN_REALM_CREATION=False):
+            # Check realm creation page is accessible
+            result = self.client.get(generated_link)
+            self.assertEquals(result.status_code, 200)
+            self.assertIn("Let's get started…", result.content)
+
+            # Create Realm with generated link
+            self.assertIsNone(get_realm(domain))
+            result = self.client.post(generated_link, {'email': email})
+            self.assertEquals(result.status_code, 302)
+            self.assertTrue(result["Location"].endswith(
+                    "/accounts/send_confirm/%s@%s" % (username, domain)))
+            result = self.client.get(result["Location"])
+            self.assertIn("Check your email so we can get started.", result.content)
+
+            # Generated link is expired
+            result = self.client.get(generated_link)
+            self.assertEquals(result.status_code, 200)
+            self.assertIn("The organization creation link has been expired or is not valid.", result.content)
+
+    def test_realm_creation_with_random_link(self):
+        with self.settings(OPEN_REALM_CREATION=False):
+            # Realm creation attempt with an invalid link should fail
+            random_link = "/create_realm/5e89081eb13984e0f3b130bf7a4121d153f1614b"
+            result = self.client.get(random_link)
+            self.assertEquals(result.status_code, 200)
+            self.assertIn("The organization creation link has been expired or is not valid.", result.content)
```

Comment:
Maybe also add a small test that the time expiration works (can make this work by setting the date of the created event back in time).